packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1396,8 +1389,9 @@ abstract class FlutterCommand extends Command<void> {
         if (_usesFatalWarnings) {
           globals.logger.fatalWarnings = boolArg(FlutterOptions.kFatalWarnings);
         }
-        // Prints the welcome message if needed.
-        globals.flutterUsage.printWelcome();
+        if (globals.analytics.shouldShowMessage) {
+          globals.analytics.clientShowedMessage();
```

Comment:
When did we show the client the message?

Reply:
Doh. That's what I get for deleting `first_run.dart` without thinking "is there a replacement for this?"

Reply:
Fixed in [restore printing analytics welcome on first run](https://github.com/flutter/flutter/pull/151518/commits/f11a25d7f0f9d61fcffa0b713b688fa64bc7257d)

Reply:
@andrewkolos to verify whether we were already printing the message from `package:unified_analytics`. If so, this PR might be making it print twice. If not, we weren't showing the message and that's not good.

Reply:
We show the consent message upon exiting the tool:

https://github.com/flutter/flutter/blob/19d0d7cffc3ab244d169a33c6a14cd234ffc7464/packages/flutter_tools/lib/src/base/process.dart#L660-L669

Addressed in commit [remove unnecessarily added consent message, add test, add newline before consent message](https://github.com/flutter/flutter/pull/151518/commits/19d0d7cffc3ab244d169a33c6a14cd234ffc7464)

Reply:
thanks!